src/components/Accordion/AccordionItemDefer.test.jsx

```diff
@@ -1,37 +1,47 @@
 import React from 'react';
-import { render, fireEvent } from '@testing-library/react';
+import { render, fireEvent, screen } from '@testing-library/react';
 
 import AccordionItemDefer from './AccordionItemDefer';
 
 import { Accordion } from '.';
 
 describe('AccordionItemDefer', () => {
   it('renders content when expanded', () => {
-    const { container, getByTestId } = render(
+    const { container } = render(
       <Accordion>
         <AccordionItemDefer id="a" title="Title one">
           <p>This content</p>
         </AccordionItemDefer>
       </Accordion>
     );
-    expect(getByTestId('accordion-item-deferred').lastElementChild.childElementCount).toEqual(0);
```

Comment:
This data-testid is on a React component. When would be an appropriate use for the way we had it over the screen query method if not this example?

Reply:
For cases where you can't match by role or text or it doesn't make sense (e.g. the text is dynamic). We should instead try to use queries that reflect the experience of visual/mouse users as well as those that use assistive technology - `getByRole`, `getByLabelText`, etc.
Testing library docs have a good overview of query preference - https://testing-library.com/docs/guide-which-query

Also, while the destructure assignments `{ container, getByTestId }` are available and present in the documentation, the emerging best practice is to query using `screen`. [The eslint rule has further info](https://github.com/testing-library/eslint-plugin-testing-library/blob/master/docs/rules/prefer-screen-queries.md) on why screen is considered best practice:

> _DOM Testing Library (and other Testing Library frameworks built on top of it) exports a screen object which has every query (and a debug method). This works better with autocomplete and makes each test a little simpler to write and maintain._